state/execution.go

```diff
@@ -123,14 +129,47 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		maxReapBytes = -1
 	}
 
+	// check if oracle's gossipVoteMap has any results
+	blockExec.oracleInfo.GossipVoteBuffer.UpdateMtx.RLock()
+	oracleVotesBuffer := blockExec.oracleInfo.GossipVoteBuffer.Buffer
+	blockExec.oracleInfo.GossipVoteBuffer.UpdateMtx.RUnlock()
+
+	var CreateOracleResultTxBz []byte
+	if len(oracleVotesBuffer) > 0 {
+		votes := []*oracleproto.GossipedVotes{}
+		for _, vote := range oracleVotesBuffer {
```

Comment:
does a copy not work..? do we even need a copy?

```
votes := make([]*oracleproto.GossipedVotes{}, len(oracleVotesBuffer))
copy(votes, oracleVotesBuffer)
```

Reply:
oracleVotesBuffer is of type map[string]GossipedVotes, so unable to copy straight